File: src/main/java/com/wolt/domain/deliveryfeecalculator/CartTotalCalculator.java
package com.wolt.domain.deliveryfeecalculator;

import java.math.BigInteger;

class CartTotalCalculator {
    private final static BigInteger REQUIRED_CART_VALUE = BigInteger.valueOf(10_00);
    private final static BigInteger NO_ADDITIONAL_FEE = BigInteger.ZERO;

    BigInteger calculate(BigInteger cartValue) {
        BigInteger cartValueDifference = REQUIRED_CART_VALUE.subtract(cartValue);
        return cartValueDifference.max(NO_ADDITIONAL_FEE);
    }
}

File: src/main/java/com/wolt/domain/deliveryfeecalculator/CartTotalCalculator.kt
package com.wolt.domain.deliveryfeecalculator

import java.math.BigInteger

internal class CartTotalCalculator {
    fun calculate(cartValue: BigInteger): BigInteger {
        val cartValueDifference = REQUIRED_CART_VALUE.subtract(cartValue)
        return cartValueDifference.max(NO_ADDITIONAL_FEE)
    }

    companion object {
        private val REQUIRED_CART_VALUE = BigInteger.valueOf(10_00)
        private val NO_ADDITIONAL_FEE = BigInteger.ZERO
    }
}